In [10]:
import os
import csv
import numpy as np
from matplotlib import pyplot as plt

fold_path = "OHIE_Data_CSV"
datafile = open('Data.csv', newline='')
dataDict = csv.DictReader(datafile, delimiter=",")
print("Great! Data is loaded")

Great! Data is loaded


In [2]:
dataDict.fieldnames

['person_id',
 'approved_app',
 'female_list',
 'age_inp',
 'health_last12_inp',
 'health_change_inp',
 'dia_dx_pre_lottery_inp',
 'dia_dx_post_lottery_inp',
 'hbp_dx_pre_lottery_inp',
 'hbp_dx_post_lottery_inp',
 'chl_dx_pre_lottery_inp',
 'chl_dx_post_lottery_inp',
 'needmet_med_inp',
 'needmet_ment_inp',
 'needmet_rx_inp',
 'med_qual_inp',
 'chl_chk_inp',
 'fobt_chk_inp',
 'cvd_risk_point_inp',
 'doc_num_incl_probe_inp',
 'ed_any_incl_probe_inp',
 'hosp_num_incl_probe_inp',
 'ins_any_inp',
 'ins_ohp_inp',
 'ins_private_inp',
 'a1c_inp',
 'chl_inp',
 'bmi_inp',
 'bp_sar_inp',
 'bp_dar_inp',
 'rx_any_mod_inp',
 'rx_num_mod_inp',
 'hbp_diure_med_inp',
 'antihyperlip_med_inp',
 'diabetes_med_inp',
 'meds_miss_inp',
 'num_visit_pre_cens_ed',
 'num_visit_cens_ed',
 'num_hiun_pre_cens_ed',
 'num_hiun_cens_ed',
 'num_loun_pre_cens_ed',
 'num_loun_cens_ed']

In [5]:
records = [ row for row in dataDict]

In [6]:
for i in range(len(records)):
    x = records[i]
    if x["approved_app"] != "":
        x["approved_app"] = int(x["approved_app"])

# Quality

In [7]:
control = []
treat = []

for i in range(len(records)):
    x = records[i]
    
    if x["approved_app"] == 1:
        treat.append(x)
    else:
        control.append(x)
print(f"There are {len(control)} people in control group")
print(f"There are {len(treat)} people in treatment group")

There are 8044 people in control group
There are 1706 people in treatment group


In [8]:
#needmet_med_inp = Got all needed medical care - Subjective
control_needmet_med_inp = [int(x["needmet_med_inp"]) for x in control if x["needmet_med_inp"] != "" ]
treat_needmet_med_inp = [int(x["needmet_med_inp"]) for x in treat if x["needmet_med_inp"] != "" ]
mean_control = np.mean(control_needmet_med_inp)
mean_treat = np.mean(treat_needmet_med_inp)
diff = mean_treat - mean_control
print(f"The mean of needmet_med_inp for control {np.mean(control_needmet_med_inp)}")
print(f"The mean of needmet_med_inp for treat {np.mean(treat_needmet_med_inp)}")
print(f"The difference {diff}")
print(f"The difference in percentages : {round(((diff/mean_control)*100),3)}%")
print("Either 0 or 1 || Yes or No question")

The mean of needmet_med_inp for control 0.6004728132387707
The mean of needmet_med_inp for treat 0.6625586854460094
The difference 0.06208587220723871
The difference in percentages : 10.339%
Either 0 or 1 || Yes or No question


In [9]:
#needmet_ment_inp = Got all needed mental health care - Subjective
control_needmet_ment_inp = [int(x["needmet_ment_inp"]) for x in control if x["needmet_ment_inp"] != "" ]
treat_needmet_ment_inp = [int(x["needmet_ment_inp"]) for x in treat if x["needmet_ment_inp"] != "" ]
mean_control = np.mean(control_needmet_ment_inp)
mean_treat = np.mean(treat_needmet_med_inp)
diff = mean_treat - mean_control
print(f"The mean of needmet_ment_inp for control {np.mean(control_needmet_ment_inp)}")
print(f"The mean of needmet_ment_inp for treat {np.mean(treat_needmet_med_inp)}")
print(f"The difference {diff}")
print(f"The difference in percentages : {round(((diff/mean_control)*100),3)}%")
print("Either 0 or 1 || Yes or No question")

The mean of needmet_ment_inp for control 0.7562009223482488
The mean of needmet_ment_inp for treat 0.6625586854460094
The difference -0.09364223690223938
The difference in percentages : -12.383%
Either 0 or 1 || Yes or No question


In [10]:
#needmet_rx_inp = Got all needed prescription medications - Subjective
control_needmet_rx_inp = [int(x["needmet_rx_inp"]) for x in control if x["needmet_rx_inp"] != "" ]
treat_needmet_rx_inp = [int(x["needmet_rx_inp"]) for x in treat if x["needmet_rx_inp"] != "" ]
mean_control = np.mean(control_needmet_rx_inp)
mean_treat = np.mean(treat_needmet_rx_inp)
diff = mean_treat - mean_control
print(f"The mean of needmet_rx_inp for control {np.mean(control_needmet_rx_inp)}")
print(f"The mean of needmet_rx_inp for treat {np.mean(treat_needmet_rx_inp)}")
print(f"The difference {diff}")
print(f"The difference in percentages : {round(((diff/mean_control)*100),3)}%")
print("Either 0 or 1 || Yes or No question")

The mean of needmet_rx_inp for control 0.7289021657953697
The mean of needmet_rx_inp for treat 0.7749120750293084
The difference 0.04600990923393866
The difference in percentages : 6.312%
Either 0 or 1 || Yes or No question


In [11]:
#med_qual_inp = Quality of medical care
control_med_qual_inp = [int(x["med_qual_inp"].split(":")[0]) for x in control if x["med_qual_inp"] != "" ]
treat_med_qual_inp = [int(x["med_qual_inp"].split(":")[0]) for x in treat if x["med_qual_inp"] != "" ]
mean_control = np.mean(control_med_qual_inp)
mean_treat = np.mean(treat_med_qual_inp)
diff = mean_treat - mean_control
print(f"The mean of needmet_rx_inp for control {mean_control}")
print(f"The mean of needmet_rx_inp for treat {mean_treat}")
print(f"The difference {diff}")
print(f"The difference in percentages : {round(((diff/mean_control)*100),3)}%")

The mean of needmet_rx_inp for control 2.622743229689067
The mean of needmet_rx_inp for treat 3.0754716981132075
The difference 0.4527284684241404
The difference in percentages : 17.262%


**Evaluation-1** <br>
1. needmet_med_inp, Got all needed medical care - Subjective <br>
Treatment group: + **10.4%** <br>
<br>
2. needmet_rx_inp, Got all needed prescription medications - Subjective <br>
Treatment group: + **6.31%** <br>
<br>
3. med_qual_inp, Quality of medical care - Subjective <br>
Treatment group: + **17.262%** <br>

# Quantity

In [12]:
#doc_num_incl_probe_inp = Number of doctor visits in the past 12 months
control_doc_num_incl_probe_inp = [int(x["doc_num_incl_probe_inp"]) for x in control if x["doc_num_incl_probe_inp"] != ""]
treat_doc_num_incl_probe_inp = [int(x["doc_num_incl_probe_inp"]) for x in treat if x["doc_num_incl_probe_inp"] != ""]
mean_control = np.mean(control_doc_num_incl_probe_inp)
mean_treat = np.mean(treat_doc_num_incl_probe_inp)
diff = mean_treat - mean_control
print(f"The mean of control_doc_num_incl_probe_inp for control {np.mean(control_doc_num_incl_probe_inp)}")
print(f"The mean of treat_doc_num_incl_probe_inp for treat {np.mean(treat_doc_num_incl_probe_inp)}")
print(f"The difference {diff}")
print(f"The difference in percentages : {round(((diff/mean_control)*100),3)}%")

The mean of control_doc_num_incl_probe_inp for control 6.341220423412204
The mean of treat_doc_num_incl_probe_inp for treat 8.037580739870815
The difference 1.6963603164586116
The difference in percentages : 26.751%


In [13]:
#ed_num_incl_probe_inp = Number of ER visits in the past 12 months
control_ed_num_incl_probe_inp = [int(x["ed_num_incl_probe_inp"]) for x in control if x["ed_num_incl_probe_inp"] != ""]
treat_ed_num_incl_probe_inp = [int(x["ed_num_incl_probe_inp"]) for x in treat if x["ed_num_incl_probe_inp"] != ""]
mean_control = np.mean(control_ed_num_incl_probe_inp)
mean_treat = np.mean(treat_ed_num_incl_probe_inp)
diff = mean_treat - mean_control
print(f"The mean of ed_num_incl_probe_inp for control {np.mean(control_ed_num_incl_probe_inp)}")
print(f"The mean of ed_num_incl_probe_inp for treat {np.mean(treat_ed_num_incl_probe_inp)}")
print(f"The difference {diff}")
print(f"The difference in percentages : {round(((diff/mean_control)*100),3)}%")

KeyError: 'ed_num_incl_probe_inp'

In [14]:
#hosp_num_incl_probe_inp = Number of hospitalizations in the past 12 months, stay in a hospital at least overnight
control_hosp_num_incl_probe_inp = [int(x["hosp_num_incl_probe_inp"]) for x in control if x["hosp_num_incl_probe_inp"] != ""]
treat_hosp_num_incl_probe_inp= [int(x["hosp_num_incl_probe_inp"]) for x in treat if x["hosp_num_incl_probe_inp"] != ""]
mean_control = np.mean(control_hosp_num_incl_probe_inp)
mean_treat = np.mean(treat_hosp_num_incl_probe_inp)
diff = mean_treat - mean_control
print(f"The mean of hosp_num_incl_probe_inp for control {np.mean(control_hosp_num_incl_probe_inp)}")
print(f"The mean of hosp_num_incl_probe_inp for treat {np.mean(treat_hosp_num_incl_probe_inp)}")
print(f"The difference {diff}")
print(f"The difference in percentages : {round(((diff/mean_control)*100),3)}%")

The mean of hosp_num_incl_probe_inp for control 0.24097135740971357
The mean of hosp_num_incl_probe_inp for treat 0.3458602466236054
The difference 0.10488888921389186
The difference in percentages : 43.528%


In [15]:
#cvd_risk_point_inp = Cardiovascular Risk, Framingham risk score
"""
This variable is constructed from the survey respondent's age, cholesterol measurements, A1c measurements, and
information on whether or not the respondent is a smoker or taking high blood pressure medication.
Cholesterol measurements
A1c measurements

"""
control_cvd_risk_point_inp = [float(x["cvd_risk_point_inp"]) for x in control if x["cvd_risk_point_inp"] != ""]
treat_cvd_risk_point_inp = [float(x["cvd_risk_point_inp"]) for x in treat if x["cvd_risk_point_inp"] != ""]
mean_control = np.mean(control_cvd_risk_point_inp)
mean_treat = np.mean(treat_cvd_risk_point_inp)
print(f"The mean of cvd_risk_point_inp for control {mean_control}")
print(f"The mean of cvd_risk_point_inp for treat {mean_treat}")
print(f"The difference = {(mean_treat-mean_control)}")
print(f"The percentage difference = {round((((mean_treat-mean_control)/mean_control) * 100),2)}%")

The mean of cvd_risk_point_inp for control 0.0805759357429719
The mean of cvd_risk_point_inp for treat 0.08500164874551971
The difference = 0.004425713002547821
The percentage difference = 5.49%


**Evaluation-2** <br>
1. doc_num_incl_probe_inp = Number of doctor visits in the past 12 months <br>
Treatment Group: + **26.751%**
<br>

2. hosp_num_incl_probe_inp = Number of hospitalizations in the past 12 months <br>
Treatment Group: + **43.528%**
<br>

3. cvd_risk_point_inp = Cardiovascular Risk, Framingham risk score <br>
Treatment Group: + **5.49%**
<br>


**Conclusion**
People who are in the treatment group was able to access the medical care and needed prescription medications more but not significantly. <br>
People who are in the treatment group was able to see the doctor and they were significantly more hospitalized, which indicates that they 

# Further investigation for obese people

In [16]:
obese_threshold = 30

obese_control = []
obese_treat = []

for i in range(len(control)):
    bmi_val = control[i]["bmi_inp"]
    
    if bmi_val and float(bmi_val) >= 30:
        obese_control.append(control[i])
        

for i in range(len(treat)):
    bmi_val = treat[i]["bmi_inp"]
    
    if bmi_val and float(bmi_val) >= 30:
        obese_treat.append(treat[i])
        
print(len(obese_control))
print(len(obese_treat))

3314
684


In [17]:
#doc_num_incl_probe_inp = Number of doctor visits in the past 12 months
control_doc_num_incl_probe_inp = [int(x["doc_num_incl_probe_inp"]) for x in obese_control if x["doc_num_incl_probe_inp"] != ""]
treat_doc_num_incl_probe_inp = [int(x["doc_num_incl_probe_inp"]) for x in obese_treat if x["doc_num_incl_probe_inp"] != ""]
mean_control = np.mean(control_doc_num_incl_probe_inp)
mean_treat = np.mean(treat_doc_num_incl_probe_inp)
diff = mean_treat - mean_control
print(f"The mean of control_doc_num_incl_probe_inp for control {np.mean(control_doc_num_incl_probe_inp)}")
print(f"The mean of treat_doc_num_incl_probe_inp for treat {np.mean(treat_doc_num_incl_probe_inp)}")
print(f"The difference {diff}")
print(f"The difference in percentages : {round(((diff/mean_control)*100),3)}%")

The mean of control_doc_num_incl_probe_inp for control 7.55535390199637
The mean of treat_doc_num_incl_probe_inp for treat 8.872620790629576
The difference 1.3172668886332053
The difference in percentages : 17.435%


In [18]:
#hosp_num_incl_probe_inp = Number of hospitalizations in the past 12 months, stay in a hospital at least overnight
control_hosp_num_incl_probe_inp = [int(x["hosp_num_incl_probe_inp"]) for x in obese_control if x["hosp_num_incl_probe_inp"] != ""]
treat_hosp_num_incl_probe_inp= [int(x["hosp_num_incl_probe_inp"]) for x in obese_treat if x["hosp_num_incl_probe_inp"] != ""]
mean_control = np.mean(control_hosp_num_incl_probe_inp)
mean_treat = np.mean(treat_hosp_num_incl_probe_inp)
diff = mean_treat - mean_control
print(f"The mean of hosp_num_incl_probe_inp for control {np.mean(control_hosp_num_incl_probe_inp)}")
print(f"The mean of hosp_num_incl_probe_inp for treat {np.mean(treat_hosp_num_incl_probe_inp)}")
print(f"The difference {diff}")
print(f"The difference in percentages : {round(((diff/mean_control)*100),3)}%")

The mean of hosp_num_incl_probe_inp for control 0.24947081947384336
The mean of hosp_num_incl_probe_inp for treat 0.2547584187408492
The difference 0.005287599267005838
The difference in percentages : 2.12%


In [19]:
#cvd_risk_point_inp = Cardiovascular Risk, Framingham risk score
"""
This variable is constructed from the survey respondent's age, cholesterol measurements, A1c measurements, and
information on whether or not the respondent is a smoker or taking high blood pressure medication.
Cholesterol measurements
A1c measurements

"""
control_cvd_risk_point_inp = [float(x["cvd_risk_point_inp"]) for x in obese_control if x["cvd_risk_point_inp"] != ""]
treat_cvd_risk_point_inp = [float(x["cvd_risk_point_inp"]) for x in obese_treat if x["cvd_risk_point_inp"] != ""]
mean_control = np.mean(control_cvd_risk_point_inp)
mean_treat = np.mean(treat_cvd_risk_point_inp)
print(f"The mean of cvd_risk_point_inp for control {mean_control}")
print(f"The mean of cvd_risk_point_inp for treat {mean_treat}")
print(f"The difference = {(mean_treat-mean_control)}")
print(f"The percentage difference = {round((((mean_treat-mean_control)/mean_control) * 100),2)}%")

The mean of cvd_risk_point_inp for control 0.08404596226415094
The mean of cvd_risk_point_inp for treat 0.08604957983193279
The difference = 0.0020036175677818485
The percentage difference = 2.38%


In [20]:
#needmet_med_inp = Got all needed medical care - Subjective
control_needmet_med_inp = [int(x["needmet_med_inp"]) for x in obese_control if x["needmet_med_inp"] != "" ]
treat_needmet_med_inp = [int(x["needmet_med_inp"]) for x in obese_treat if x["needmet_med_inp"] != "" ]
mean_control = np.mean(control_needmet_med_inp)
mean_treat = np.mean(treat_needmet_med_inp)
diff = mean_treat - mean_control
print(f"The mean of needmet_med_inp for control {np.mean(control_needmet_med_inp)}")
print(f"The mean of needmet_med_inp for treat {np.mean(treat_needmet_med_inp)}")
print(f"The difference {diff}")
print(f"The difference in percentages : {round(((diff/mean_control)*100),3)}%")
print("Either 0 or 1 || Yes or No question")

The mean of needmet_med_inp for control 0.575226586102719
The mean of needmet_med_inp for treat 0.6471449487554904
The difference 0.0719183626527714
The difference in percentages : 12.503%
Either 0 or 1 || Yes or No question


In [21]:
#needmet_rx_inp = Got all needed prescription medications - Subjective
control_needmet_rx_inp = [int(x["needmet_rx_inp"]) for x in obese_control if x["needmet_rx_inp"] != "" ]
treat_needmet_rx_inp = [int(x["needmet_rx_inp"]) for x in obese_treat if x["needmet_rx_inp"] != "" ]
mean_control = np.mean(control_needmet_rx_inp)
mean_treat = np.mean(treat_needmet_rx_inp)
diff = mean_treat - mean_control
print(f"The mean of needmet_rx_inp for control {np.mean(control_needmet_rx_inp)}")
print(f"The mean of needmet_rx_inp for treat {np.mean(treat_needmet_rx_inp)}")
print(f"The difference {diff}")
print(f"The difference in percentages : {round(((diff/mean_control)*100),3)}%")
print("Either 0 or 1 || Yes or No question")

The mean of needmet_rx_inp for control 0.6795646916565901
The mean of needmet_rx_inp for treat 0.7456140350877193
The difference 0.06604934343112923
The difference in percentages : 9.719%
Either 0 or 1 || Yes or No question


In [22]:
#med_qual_inp = Quality of medical care
control_med_qual_inp = [int(x["med_qual_inp"].split(":")[0]) for x in obese_control if x["med_qual_inp"] != "" ]
treat_med_qual_inp = [int(x["med_qual_inp"].split(":")[0]) for x in obese_treat if x["med_qual_inp"] != "" ]
mean_control = np.mean(control_med_qual_inp)
mean_treat = np.mean(treat_med_qual_inp)
diff = mean_treat - mean_control
print(f"The mean of needmet_rx_inp for control {mean_control}")
print(f"The mean of needmet_rx_inp for treat {mean_treat}")
print(f"The difference {diff}")
print(f"The difference in percentages : {round(((diff/mean_control)*100),3)}%")
print(len(control_med_qual_inp))
print(len(treat_med_qual_inp))

The mean of needmet_rx_inp for control 2.7568472306755933
The mean of needmet_rx_inp for treat 3.225806451612903
The difference 0.4689592209373097
The difference in percentages : 17.011%
3286
682


**Evaluation-3** <br>
1. needmet_med_inp, Got all needed medical care - Subjective <br>
Treatment group: + **10.4% -> 12.503%** <br>
<br>
2. needmet_rx_inp, Got all needed prescription medications - Subjective <br>
Treatment group: + **6.31% -> 9.719%** <br>
<br>
3. med_qual_inp, Quality of medical care - Subjective <br>
Treatment group: + **17.262% -> 17.011%** <br>
<br>

4. doc_num_incl_probe_inp = Number of doctor visits in the past 12 months <br>
Treatment Group: + **26.751% -> 17.435%**
<br>

5. hosp_num_incl_probe_inp = Number of hospitalizations in the past 12 months <br>
Treatment Group: + **43.528% -> 2.12%**
<br>

6. cvd_risk_point_inp = Cardiovascular Risk, Framingham risk score <br>
Treatment Group: + **5.49% -> 2.38%**
<br>


# Conclusion
Maybe the increase in percentages seem positive change. However, the numbers are quite low

In [11]:
records = [ row for row in dataDict]
for i in range(len(records)):
    x = records[i]
    if x["approved_app"] != "":
        x["approved_app"] = int(x["approved_app"])

In [12]:
control = []
treat = []

for i in range(len(records)):
    x = records[i]
    
    if x["approved_app"] == 1:
        treat.append(x)
    elif (x["ins_any_inp"] == "" or x["ins_any_inp"] == "0") and (x["ins_ohp_inp"] == "" or x["ins_ohp_inp"] == "0") and (x["ins_private_inp"] == "" or x["ins_private_inp"] == "0"):
        control.append(x)
print(f"There are {len(control)} people in control group")
print(f"There are {len(treat)} people in treatment group")

There are 5977 people in control group
There are 1706 people in treatment group


In [13]:
# rx_num_mod_inp Number of prescription medications currently taking (modified, see codebook for
# hbp_dirue_med_inp Taking any medication for high blood pressure medication (ETC code)
# antihyperlip_med_inp Taking at least one high cholesterol medication (ETC code)
# diabetes_med_inp Taking at least one diabetes medication (ETC code)

In [6]:
control[0]

{'person_id': '8',
 'approved_app': 0,
 'female_list': '0',
 'age_inp': '41',
 'health_last12_inp': '3: Fair',
 'health_change_inp': '1: Worse',
 'dia_dx_pre_lottery_inp': '0',
 'dia_dx_post_lottery_inp': '0',
 'hbp_dx_pre_lottery_inp': '0',
 'hbp_dx_post_lottery_inp': '1',
 'chl_dx_pre_lottery_inp': '0',
 'chl_dx_post_lottery_inp': '1',
 'needmet_med_inp': '0',
 'needmet_ment_inp': '1',
 'needmet_rx_inp': '0',
 'med_qual_inp': '2: Fair',
 'chl_chk_inp': '1',
 'fobt_chk_inp': '',
 'cvd_risk_point_inp': '0.112',
 'doc_num_incl_probe_inp': '6',
 'ed_any_incl_probe_inp': '1',
 'hosp_num_incl_probe_inp': '0',
 'ins_any_inp': '0',
 'ins_ohp_inp': '0',
 'ins_private_inp': '0',
 'a1c_inp': '5.2005',
 'chl_inp': '229.859',
 'bmi_inp': '35.2312',
 'bp_sar_inp': '134',
 'bp_dar_inp': '82',
 'rx_any_mod_inp': '1',
 'rx_num_mod_inp': '2',
 'hbp_diure_med_inp': '0',
 'antihyperlip_med_inp': '0',
 'diabetes_med_inp': '0',
 'meds_miss_inp': '0',
 'num_visit_pre_cens_ed': '0',
 'num_visit_cens_ed': '2

In [14]:
# rx_any_mod_inp Curently taking any prescription medications (modified, see codebook)
control_rx_any_mod_inp = []
treat_rx_any_mod_inp = []

for t in control:
    if t["rx_any_mod_inp"] == "":
        continue
    if t["rx_any_mod_inp"] == "1":
        control_rx_any_mod_inp.append(1)
    else:
        control_rx_any_mod_inp.append(0)
        
for t in treat:
    if t["rx_any_mod_inp"] == "":
        continue
    if t["rx_any_mod_inp"] == "1":
        treat_rx_any_mod_inp.append(1)
    else:
        treat_rx_any_mod_inp.append(0)

mean_control = np.mean(control_rx_any_mod_inp)
mean_treat = np.mean(treat_rx_any_mod_inp)
diff = mean_treat - mean_control
print(f"The mean of rx_any_mod_inp for control {mean_control}")
print(f"The mean of rx_any_mod_inp for treat {mean_treat}")
print(f"The difference {diff}")
print(f"The difference in percentages : {round(((diff/mean_control)*100),3)}%")

The mean of rx_any_mod_inp for control 0.44929718875502006
The mean of rx_any_mod_inp for treat 0.6178194607268465
The difference 0.1685222719718264
The difference in percentages : 37.508%


In [18]:

control_rx_num_mod_inp = []
treat_rx_num_mod_inp = []

for t in control:
    if t["rx_num_mod_inp"] == "":
        continue
    control_rx_num_mod_inp.append(float(t["rx_num_mod_inp"]))
        
for t in treat:
    if t["rx_num_mod_inp"] == "":
        continue
    treat_rx_num_mod_inp.append(float(t["rx_num_mod_inp"]))
    
mean_control = np.mean(control_rx_num_mod_inp)
mean_treat = np.mean(treat_rx_num_mod_inp)
diff = mean_treat - mean_control
print(f"The mean of rx_num_mod_inp for control {mean_control}")
print(f"The mean of rx_num_mod_inp for treat {mean_treat}")
print(f"The difference {diff}")
print(f"The difference in percentages : {round(((diff/mean_control)*100),3)}%")

The mean of rx_num_mod_inp for control 1.2187124463519314
The mean of rx_num_mod_inp for treat 2.391826923076923
The difference 1.1731144767249915
The difference in percentages : 96.259%


In [15]:
control_hbp_diure_med_inp = []
treat_hbp_diure_med_inp = []

for t in control:
    if t["hbp_diure_med_inp"] == "":
        continue
    if t["hbp_diure_med_inp"] == "1":
        control_hbp_diure_med_inp.append(1)
    else:
        control_hbp_diure_med_inp.append(0)
        
for t in treat:
    if t["hbp_diure_med_inp"] == "":
        continue
    if t["hbp_diure_med_inp"] == "1":
        treat_hbp_diure_med_inp.append(1)
    else:
        treat_hbp_diure_med_inp.append(0)

mean_control = np.mean(control_hbp_diure_med_inp)
mean_treat = np.mean(treat_hbp_diure_med_inp)
diff = mean_treat - mean_control
print(f"The mean of hbp_diure_med_inp for control {mean_control}")
print(f"The mean of hbp_diure_med_inp for treat {mean_treat}")
print(f"The difference {diff}")
print(f"The difference in percentages : {round(((diff/mean_control)*100),3)}%")

The mean of hbp_diure_med_inp for control 0.09804249623556968
The mean of hbp_diure_med_inp for treat 0.1623681125439625
The difference 0.06432561630839281
The difference in percentages : 65.61%


In [16]:
control_antihyperlip_med_inp = []
treat_hantihyperlip_med_inp = []

for t in control:
    if t["antihyperlip_med_inp"] == "":
        continue
    if t["antihyperlip_med_inp"] == "1":
        control_antihyperlip_med_inp.append(1)
    else:
        control_antihyperlip_med_inp.append(0)
        
for t in treat:
    if t["antihyperlip_med_inp"] == "":
        continue
    if t["antihyperlip_med_inp"] == "1":
        treat_hantihyperlip_med_inp.append(1)
    else:
        treat_hantihyperlip_med_inp.append(0)

mean_control = np.mean(control_antihyperlip_med_inp)
mean_treat = np.mean(treat_hantihyperlip_med_inp)
diff = mean_treat - mean_control
print(f"The mean of antihyperlip_med_inp for control {mean_control}")
print(f"The mean of antihyperlip_med_inp for treat {mean_treat}")
print(f"The difference {diff}")
print(f"The difference in percentages : {round(((diff/mean_control)*100),3)}%")

The mean of antihyperlip_med_inp for control 0.05956165300317885
The mean of antihyperlip_med_inp for treat 0.10492379835873387
The difference 0.04536214535555502
The difference in percentages : 76.16%


In [17]:
control_diabetes_med_inp = []
treat_diabetes_med_inp = []

for t in control:
    if t["diabetes_med_inp"] == "":
        continue
    if t["diabetes_med_inp"] == "1":
        control_diabetes_med_inp.append(1)
    else:
        control_diabetes_med_inp.append(0)
        
for t in treat:
    if t["diabetes_med_inp"] == "":
        continue
    if t["diabetes_med_inp"] == "1":
        treat_diabetes_med_inp.append(1)
    else:
        treat_diabetes_med_inp.append(0)

mean_control = np.mean(control_diabetes_med_inp)
mean_treat = np.mean(treat_diabetes_med_inp)
diff = mean_treat - mean_control
print(f"The mean of diabetes_med_inp for control {mean_control}")
print(f"The mean of diabetes_med_inp for treat {mean_treat}")
print(f"The difference {diff}")
print(f"The difference in percentages : {round(((diff/mean_control)*100),3)}%")

The mean of diabetes_med_inp for control 0.05387318052534716
The mean of diabetes_med_inp for treat 0.07268464243845252
The difference 0.018811461913105357
The difference in percentages : 34.918%


**Evaluation, Whether treatment group takes the recommended care**
<br>
1) rx_num_mod_inp Number of prescription medications currently taking <br>
 Treatment Group **+17.349**
<br>
2) hbp_dirue_med_inp Taking any medication for high blood pressure medication <br>
 Treatment Group **+29.88**
<br>
3) diabetes_med_inp Taking at least one diabetes medication <br>
 Treatment Group **+12.318%**
<br>

# All evaluations

**Quality- General**<br>
1. needmet_med_inp, Got all needed medical care - Subjective <br>
Treatment group: + **10.4%** <br>
<br>
2. needmet_rx_inp, Got all needed prescription medications - Subjective <br>
Treatment group: + **6.31%** <br>
<br>
3. med_qual_inp, Quality of medical care - Subjective <br>
Treatment group: + **17.262%** <br>

**Quantity - General** <br>
1. doc_num_incl_probe_inp = Number of doctor visits in the past 12 months <br>
Treatment Group: + **26.751%**
<br>

2. hosp_num_incl_probe_inp = Number of hospitalizations in the past 12 months <br>
Treatment Group: + **43.528%**
<br>

3. cvd_risk_point_inp = Cardiovascular Risk, Framingham risk score <br>
Treatment Group: + **5.49%**
<br>

**Quality & Quantity for Obese People w/ Comparisons** <br>
1. needmet_med_inp, Got all needed medical care - Subjective <br>
Treatment group: + **10.4% -> 12.503%** <br>
<br>
2. needmet_rx_inp, Got all needed prescription medications - Subjective <br>
Treatment group: + **6.31% -> 9.719%** <br>
<br>
3. med_qual_inp, Quality of medical care - Subjective <br>
Treatment group: + **17.262% -> 17.011%** <br>
<br>

4. doc_num_incl_probe_inp = Number of doctor visits in the past 12 months <br>
Treatment Group: + **26.751% -> 17.435%**
<br>

5. hosp_num_incl_probe_inp = Number of hospitalizations in the past 12 months <br>
Treatment Group: + **43.528% -> 2.12%**
<br>

6. cvd_risk_point_inp = Cardiovascular Risk, Framingham risk score <br>
Treatment Group: + **5.49% -> 2.38%**
<br>


**Quantity check whether treatment group takes the recommended care**
<br>
1) rx_num_mod_inp Number of prescription medications currently taking <br>
 Treatment Group **+17.349**
<br>
2) hbp_dirue_med_inp Taking any medication for high blood pressure medication <br>
 Treatment Group **+29.88**
<br>
3) diabetes_med_inp Taking at least one diabetes medication <br>
 Treatment Group **+12.318%**
<br>

**QUESTIONs & IDEAS**
<br>
- Linear model to see if the being diagnosed increased the likelihood of taking medication?

In [23]:
(9.54-6.91)/6.91 *100

38.06078147612155